### PR TITLE
refactor(components): [descriptions] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by ryuhangyeong**
## Description

#6114 

---

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/descriptions/src/index.vue [hunk 1]
```diff
@@ -24,128 +24,119 @@
...
 import type { ComponentSize } from '@element-plus/constants'
 
-export default defineComponent({
+defineOptions({
   name: 'ElDescriptions',
-  components: {
-    [DescriptionsRow.name]: DescriptionsRow,
+})
+
+const props = defineProps({
```
[the remark this answers is not in the document]
Comment (ryuhangyeong) [reply packages/components/descriptions/src/index.vue]: @sxzz changes to the file name (description.vue) must also change the test code. If I changed it in the PR, it crashed, so I didn't do it separately.

File: packages/components/descriptions/src/description.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  border: {
+    type: Boolean,
+    default: false,
+  },
+  column: {
+    type: Number,
+    default: 3,
+  },
+  direction: {
+    type: String as PropType<'horizontal' | 'vertical'>,
```
Comment (sxzz) [inline packages/components/descriptions/src/description.ts:17]:
```suggestion
    type: String,
    values: ['horizontal', 'vertical'],
```

File: packages/components/descriptions/src/description.vue [hunk 3]
```diff
@@ -0,0 +1,114 @@
+<template>
```
Comment (sxzz) [inline packages/components/descriptions/src/description.vue:1]: Please fix all TS error, and remove path from `internal/build/src/type-unsafe-stricter.json`

Comment (ryuhangyeong) [reply packages/components/descriptions/src/description.vue:1]: Sorry for not giving you feedback on this PR for a long time.

I tried to fix the TS error, but I'm having difficulties.

Could it be possible to only work on the switch to script-setup syntax except for the ts error in the PR? All of that work has been done.

Comment (ryuhangyeong) [reply packages/components/descriptions/src/description.vue:1]: @sxzz Sorry for calling you directly through the mention, but can you review this PR?

**Review by sxzz — APPROVED** [review]

**Pull request merged**